platform-managers/K8sPlatformManager.js

```diff
@@ -1,13 +1,29 @@
 'use strict';
 
 const BasePlatformManager = require('./BasePlatformManager');
+const Promise = require('bluebird');
 
 class K8sPlatformManager extends BasePlatformManager {
   constructor(platform) {
     super(platform);
     this.platform = platform;
   }
 
+  preUnbindOperations(options) {
+    /* jshint unused:false */
+    return Promise.resolve();
+  }
+
+  preBindOperations(options) {
```

Comment:
Ideally we follow empty base implementation in the `BasePlatformManager` and implementers doing specific things. It would make sense in this case as well. You can move the empty methods and put cf specific implementation in the `CfPlatformManager`

Reply:
The problem here is that the empty base implementation is not returning a promise and that's why the binding operations fail. I guess it's just semantics- we can return an empty resolved promise in the base (as I suggested in the comment above) and not implement anything in the K8SPlatformManager or stay with what @jayeshjk has done here. I am fine either way

Reply:
@abh1kg @subhankarc with this implementation, if the incoming request doesn't contain `context.platform`, it will be processed by `BasePlatformManager` and will fail. We can maybe take a call based on this behaviour, i.e., if we want to enforce that request should contain `context.platform`, then we could go with current way.